tests/kafkatest/services/copycat.py

```diff
@@ -0,0 +1,91 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.services.service import Service
+from ducktape.utils.util import wait_until
+from .util import monitor_log
+import subprocess, signal
+
+
+class CopycatStandaloneService(Service):
+    """Runs Copycat in standalone mode."""
+
+    logs = {
+        "kafka_log": {
+            "path": "/mnt/copycat.log",
+            "collect_default": True},
+    }
+
+    def __init__(self, context, kafka, files):
+        super(CopycatStandaloneService, self).__init__(context, 1)
+        self.kafka = kafka
+        self.files = files
+
+    def set_configs(self, config_template, connector_config_template):
+        """
+        Set configurations for the standalone worker and the connector to run on
+        it. These are not provided in the constructor because at least the
+        standalone worker config generally needs access to ZK/Kafka services to
+        create the configuration.
+        """
+        self.config_template = config_template
+        self.connector_config_template = connector_config_template
+
+    # For convenience since this service only makes sense with a single node
+    @property
+    def node(self):
+        return self.nodes[0]
+
+    def start(self):
+        super(CopycatStandaloneService, self).start()
+
+    def start_node(self, node):
+        node.account.create_file("/mnt/copycat.properties", self.config_template)
+        node.account.create_file("/mnt/copycat-connector.properties", self.connector_config_template)
+
+        self.logger.info("Starting Copycat standalone process")
+        with monitor_log(self.node, "/mnt/copycat.log") as monitor:
+            node.account.ssh("/opt/kafka/bin/copycat-standalone.sh /mnt/copycat.properties /mnt/copycat-connector.properties " +
+                             "1>> /mnt/copycat.log 2>> /mnt/copycat.log & echo $! > /mnt/copycat.pid")
+            monitor.wait_until('Copycat started', timeout_sec=10, err_msg="Never saw message indicating Copycat finished startup")
+
+        if len(self.pids()) == 0:
+            raise RuntimeError("No process ids recorded")
+
+    def pids(self):
+        """Return process ids for Copycat processes."""
+        try:
+            return [pid for pid in self.node.account.ssh_capture("cat /mnt/copycat.pid", callback=int)]
+        except:
+            return []
+
+    def stop_node(self, node, clean_shutdown=True):
+        pids = self.pids()
+        sig = signal.SIGTERM if clean_shutdown else signal.SIGKILL
+
+        for pid in pids:
+            self.node.account.signal(pid, sig, allow_fail=False)
+        for pid in pids:
+            wait_until(lambda: self.node.account.ssh("kill -0 " + str(pid), allow_fail=True) == 0, timeout_sec=10, err_msg="Copycat standalone process took too long to exit")
```

Comment:
When I tried running, this timed out.
Shouldn't it be the opposite condition: `kill -0 <pid>` is 0 iff the process exists right?
In any case, you could also try using `node.account.alive(pid)` (see ducktape RemoteAccount.alive)